tests/rados/test_crash_daemon.py

```diff
@@ -0,0 +1,108 @@
+        out, _ = cephadm.shell(args=["ceph osd ls"])
+        osd_list = out.strip().split("\n")
+        osd_id = random.choice(osd_list)
+        # ensure chosen daemon is not already in crash state
+        health_detail, _ = cephadm.shell(["ceph health detail"])
+        assert f"osd.{osd_id} crashed" not in health_detail
+        assert rados_obj.crash_ceph_daemon(daemon="osd", id=osd_id, manual_inject=True)
+
+        # 2. Crash a MON daemon and let the crash show up automatically
+        mon_list = rados_obj.run_ceph_command("ceph status")["quorum_names"]
+        mon_id = mon_list[0]
+        # ensure chosen daemon is not already in crash state
+        health_detail, _ = cephadm.shell(["ceph health detail"])
+        assert f"mon.{mon_id} crashed" not in health_detail
+        assert rados_obj.crash_ceph_daemon(daemon="mon", id=mon_id)
+
+        # logging ceph crash stat
+        out, _ = cephadm.shell(["ceph crash stat"])
+        log.info(out)
+
+        # archive all existing crashes
+        crash_id_list = [
+            x["crash_id"] for x in rados_obj.run_ceph_command("ceph crash ls-new")
+        ]
+        for crash_id in crash_id_list:
+            # log the ceph crash info before archiving
+            log.debug(f"Printing crash id: {crash_id} info")
+            crash_info, _ = cephadm.shell([f"ceph crash info {crash_id}"])
+            log.debug(crash_info)
+            cephadm.shell([f"ceph crash archive {crash_id}"])
+        time.sleep(5)
+        if len(rados_obj.run_ceph_command("ceph crash ls-new")) != 0:
+            log.error("Crash archive failed, crash ls-new still shows crashes")
+            raise AssertionError(
+                "Crash archive failed, crash ls-new still shows crashes"
+            )
+        log.info("Archiving of all all existing crashes complete")
+
+        # logging ceph crash stat
+        out, _ = cephadm.shell(["ceph crash stat"])
+        log.info(out)
+
+        # verification of crash related config parameters
+        # https://docs.ceph.com/en/latest/mgr/crash/#options
+        assert mon_obj.set_config(
+            section="mgr", name="mgr/crash/warn_recent_interval", value=1
+        )
+        assert mon_obj.set_config(
+            section="mgr", name="mgr/crash/retain_interval", value=2
+        )
+    except AssertionError as AE:
+        log.error(f"Verification failed with exception: {AE.__doc__}")
+        log.exception(AE)
+        return 1
+    finally:
+        log.info("\n\n ********* Executing finally block ******** \n\n")
+        time.sleep(30)
+        # fetch crash list
+        crash_ls_json = rados_obj.run_ceph_command("ceph crash ls")
+        # clear all the crashes
+        if crash_ls_json:
+            for crash in crash_ls_json:
+                cephadm.shell([f"ceph crash rm {crash['crash_id']}"])
+            rados_obj.change_osd_state(action="restart", target=osd_id)
+        cephadm.shell(["ceph orch restart mgr"])
```

Comment:
Any reason this is included here?

Reply:
It removes any stale warnings which could remain in `ceph health` output

Reply:
Let's raise a med bug for this? if crash warnings persist even after `crash rm`, it might cause confusions.

Reply:
No no, crash warnings do not persist, I have verified multiple times, ceph health state is back to OK almost immediately, please let me know if I should remove the mgr restart command
As this test is getting after many other prior tests, I though restarting mgr would bring the cluster health to accurate state.

Reply:
Let's leave it then. No issues.